sdk/eventhub/azure-eventhub/azure/eventhub/_consumer.py

```diff
@@ -193,14 +204,14 @@ def _open(self):
                 self.handler_ready = True
         return self.handler_ready
 
-    def receive(self):
-        # type: () -> None
+    def receive(self, batch=False, max_batch_size=300, max_wait_time=None):
         retried_times = 0
         max_retries = (
             self._client._config.max_retries  # pylint:disable=protected-access
         )
-
-        if not self._message_buffer:  # then fetch some messages into buffer
+        self._receive_start_time = self._receive_start_time or time.time()
```

Comment:
I'm not sure why this is assigned to self? Seems like it can just be a local variable?

Reply:
As per our discussion, the sync is single-threaded and all partitions are receiving alternately. So we need to record the start time of a partition on the consumer instance